File: models/implementations/gru_mean.js
/* GRU + graph mean. See matching Python module for training. */
(function(global){
'use strict';
const registry=typeof module!=='undefined'?require('../../shared/runtime/model-registry'):global.FraudAdapters;
const base=typeof module!=='undefined'?require('./_neural'):global.FraudNeuralPrimitives;
// Architecture: gru memory; mean readout; 2 graph layers.
const adapter={initialize:base.initialize,update:base.update,readout:base.readout,predict:base.predict};
registry.register('gru_mean',adapter);
if(typeof module!=='undefined')module.exports=adapter;
})(globalThis);

File: models/implementations/gru_mean.py
"""GRU + graph mean: explicit architecture using shared differentiable primitives."""
from ._neural import NeuralModel,train_design
MODEL_ID='gru_mean'

class Model(NeuralModel):
    architecture={'id': 'gru_mean', 'label': 'GRU + graph mean', 'hidden': 8, 'memory': 'gru', 'readout': 'mean', 'layers': 2, 'description': 'GRU account memory plus two layers averaging direction-, amount-, and age-aware neighbor messages.'}

def train(args,data,labeled=None):
    return train_design(args,data,MODEL_ID,labeled)
